src/main/host/descriptor/file.c

```diff
@@ -292,7 +292,7 @@ int file_openat(File* file, File* dir, const char* pathname, int flags, mode_t m
     /* The os-backed file is now ready. */
     descriptor_adjustStatus(&file->super, STATUS_DESCRIPTOR_ACTIVE, TRUE);
 
-    return _file_getFD(file);
+    return 0;
```

Comment:
Add a comment here, or document in the header file what `file_open` and `file_openat` are expected to return. (It's useful since the return logic is slightly different from `open` and `openat`.)

Reply:
I changed the return value back, but made it an error to call `file_openat()` if you haven't registered the descriptor yet.